polkadot/node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -162,9 +198,125 @@ impl Requester {
 			self.add_cores(ctx, runtime, leaf, leaf_session_index, cores).await?;
 		}
 
+		let groups = get_validator_groups(sender, new_head.hash).await?;
+		self.early_request_chunks(ctx, runtime, new_head, leaf_session_index, &groups).await;
 		Ok(())
 	}
 
+	async fn early_request_chunks<Context>(
+		&mut self,
+		ctx: &mut Context,
+		runtime: &mut RuntimeInfo,
+		activated_leaf: ActivatedLeaf,
+		leaf_session_index: SessionIndex,
+		validator_groups: &(Vec<Vec<ValidatorIndex>>, GroupRotationInfo),
+	) {
+		let sender = &mut ctx.sender().clone();
+
+		let availability_cores = match get_availability_cores(sender, activated_leaf.hash).await {
```

Comment:
nit  - IMO `inspect_err` and `ok()?` is a bit cleaner - but that might just be my personal preference:

```rust
let availability_cores = get_availability_cores(sender, activated_leaf.hash)
	.await
	.inspect_err(|err| {
		gum::warn!(
			target: LOG_TARGET,
			error = ?err,
			"Failed to get availability cores for leaf {:?}",
			activated_leaf.hash
		)
	})
	.ok()?; 
```

(and same below with `backable_candidate_hashes`)